core/main/src/broker/endpoint_broker.rs

```diff
@@ -685,24 +692,69 @@ async fn forward_extn_event(
     }
 }
 
-fn apply_response(result: Value, filter: String, rpc_request: &RpcRequest, v: &mut BrokerOutput) {
+fn apply_response(
+    raw_value: Value,
+    filter: String,
+    rpc_request: &RpcRequest,
+    v: &mut BrokerOutput,
+) {
     match jq_compile(
-        result.clone(),
+        raw_value.clone(),
         &filter,
         format!("{}_response", rpc_request.ctx.method),
     ) {
-        Ok(r) => {
-            if r.to_string().to_lowercase().contains("null") {
-                v.data.result = Some(Value::Null)
-            } else if result.get("success").is_some() {
-                v.data.result = Some(r);
-                v.data.error = None;
+        Ok(compilation_result) => {
+            debug!(
+                "jq_compile result {:?} for {}",
+                compilation_result, raw_value
+            );
+            if compilation_result == Value::Null {
+                error!(
+                    "error processing: {} from {}",
+                    compilation_result, raw_value
+                );
+                v.data.error = Some(Value::from(false));
+                v.data.result = Some(Value::Null);
+            } else if compilation_result.get("success").is_some() {
+                v.data.result = Some(compilation_result.clone());
+                v.data.error = match compilation_result.get("success") {
+                    Some(v) => {
+                        if v.is_boolean() {
+                            if !v.as_bool().unwrap() {
+                                Some(Value::from(false))
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        }
+                    }
+                    None => None,
+                };
+            } else if raw_value.get("success").is_some() {
```

Comment:
This needs to be re-checked for `result` and `error` use cases.    
`result` response struct is implementation defined and `success` field is not part of standard JSON-RPC 2.0 response format.